R/getExperimentCrossCorrelation.R

```diff
@@ -0,0 +1,577 @@
+          stop("'p_adj_threshold' must be a numeric value >= 0.", call. = FALSE)
+        }
+        # Check n_signif
+        if (!is.numeric(n_signif) && n_signif >= 0 ){
+          stop("'n_signif' must be a numeric value greater than or equal to 0.", 
+               call. = FALSE)
+        }
+        # Check cor_threshold
+        if( ifelse( is.numeric(cor_threshold), !(cor_threshold>=0 && cor_threshold<=1), !is.null(cor_threshold)) ){
+          stop("'cor_threshold' must be a numeric value greater than or equal to 0.", 
+               call. = FALSE)
+        }
+        # Check sort
+        if( !(sort == TRUE || sort == FALSE) ){
+          stop("'sort' must be a boolean value.", 
+               call. = FALSE)
+        }
+        # Check filter_self_correlations
+        if( !(filter_self_correlations == TRUE || filter_self_correlations == FALSE) ){
+          stop("'filter_self_correlations' must be a boolean value.", 
+               call. = FALSE)
+        }
+        ############################ INPUT CHECK END ###########################
+        # Fetch assays to correlate
+        assay1 <- assay(tse1, abund_values1)
+        assay2 <- assay(tse2, abund_values1)
+        # Transposes tables to right format
+        assay1 <- t(assay1)
+        assay2 <- t(assay2)
+        # Converts tables to data frame
+        assay1 <- as.data.frame(assay1)
+        assay2 <- assay2
+        
+        # Get assay in right format
+        assay1 <- .cor_test_data_type(assay1, method)
+        # Calculate correlations
+        results <- .calculate_correlation(assay1, assay2, method, p_adj_method)
+        correlations <- results$correlations
+        p_values <- results$p_values
+        p_values_adjusted <- results$p_values_adjusted
+        # Do filtering
+        result <- .correlation_filter(correlations, 
+                                      p_values, 
+                                      p_values_adjusted, 
+                                      p_adj_threshold,
+                                      n_signif, 
+                                      cor_threshold,
+                                      sort,
+                                      assay1, assay2, filter_self_correlations)
+        # Matrix or table?
+        if (mode == "table") {
+          result <- .correlation_matrix_to_table(result)
+        }
+        return(result)
+    }
+)
+
+#' @rdname getExperimentCrossCorrelation
+#' @export
+setMethod("getExperimentCrossCorrelation", signature = c(x = "TreeSummarizedExperiment"),
+    function(x,
+             experiment2 = NULL,
+             abund_values1 = "counts",
+             abund_values2 = "counts",
+             method = "spearman",
+             mode = "table",
+             p_adj_method = "fdr",
+             p_adj_threshold = 0.05,
+             n_signif = 0,
+             cor_threshold = NULL,
+             sort = FALSE,
+             filter_self_correlations = FALSE){
```

Comment:
use `...` for most of the arguments. No point in having to maintain all the args twice, when the are not used in the wrapper.